hollow/src/main/java/com/netflix/hollow/core/write/HollowWriteStateEngine.java

```diff
@@ -168,6 +171,11 @@ public void run() {
 
         previousStateRandomizedTag = readStateEngine.getCurrentRandomizedTag();
         nextStateRandomizedTag = mintNewRandomizedStateTag();
+        if (readStateEngine.getHeaderTag(HEADER_TAG_METRIC_CYCLE_START) != null) {
+            previousCycleStartTs = OptionalLong.of(Long.valueOf(
```

Comment:
I understand that we keep track of previous cycle TS. In restoreFrom(), If a producer is restarted then why do we need to capture the previous cycle TS? Isn't this value redundant in this scenario?

Reply:
Okay, this value is assigned during reverse delta computation..